src/thermo/Phase.cpp

```diff
@@ -294,10 +322,18 @@ void Phase::setMoleFractionsByName(const compositionMap& xMap)
     vector_fp mf(m_kk, 0.0);
     for (const auto& sp : xMap) {
         try {
-            mf[m_speciesIndices.at(toLowerCopy(sp.first))] = sp.second;
+            mf[m_speciesIndices.at(sp.first)] = sp.second;
         } catch (std::out_of_range&) {
-            throw CanteraError("Phase::setMoleFractionsByName",
-                               "Unknown species '{}'", sp.first);
+            size_t loc = npos;
+            if (!m_caseSensitiveSpecies) {
+                loc = findSpeciesLower(sp.first);
+            }
+            if (loc == npos) {
+                throw CanteraError("Phase::setMoleFractionsByName",
+                                   "Unknown species '{}'", sp.first);
+            } else {
+                mf[loc] = sp.second;
+            }
```

Comment:
Pending the suggested implementation change to `speciesIndex`, I think this could be simplified to just:
```c++
size_t k = speciesIndex(sp.first);
if (k != npos) {
    mf[loc] = sp.second;
} else {
    throw CanteraError(...);
}
```

Reply:
see above: `findSpeciesLower` is used as a fallback solution.

Reply:
@speth on second thought, the suggested solution will work regardless of the implementation. Adopting ... :thumbsup: